llvm/lib/Transforms/Scalar/EarlyCSE.cpp

```diff
@@ -978,7 +981,8 @@ class EarlyCSE {
         V = II->getOperand(0);
         break;
       default:
-        return TTI.getOrCreateResultFromMemIntrinsic(II, ExpectedType);
+        return TTI.getOrCreateResultFromMemIntrinsic(II, ExpectedType,
```

Comment:
Can we use `TmpInstructions` directly?

Reply:
Yes, if we make the function non-const, but that would expose much more to be potentially modified, which is why I went with passing in the SmallVectorImpl. can change if preferred